core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -342,4 +313,66 @@ public void onFailure(Throwable e) {
             }
         }
     }
+
+    /**
+     * Internal helper class to store characteristic offsets of a buffer during processing
+     */
+    protected static final class Marker {
+        private final ChannelBuffer buffer;
+        private final int size;
+        private final int markedReaderIndex;
+        private final int expectedReaderIndex;
+
+        public Marker(ChannelBuffer buffer) {
+            this.buffer = buffer;
+            this.size = buffer.getInt(buffer.readerIndex() - 4);
```

Comment:
It seems we're keeping this class. Can we add docs to the different fields and what they mean? what is "markedReaderIndex" ? what is "expectedReaderIndex"? also I think the java docs of the class can be improved (i.e., capture the request start position and expected length in order to do validation